apps/test-suite/tests/ScreenOrientation.js

```diff
@@ -0,0 +1,384 @@
+          const orientationLock = await ScreenOrientation.getOrientationLockAsync();
+          t.expect(orientationLock).toBe(ScreenOrientation.OrientationLock.OTHER);
+
+          // expect the native platform getter to return correctly
+          const platformInfo = await ScreenOrientation.getPlatformOrientationLockAsync();
+          const { screenOrientationConstantAndroid } = platformInfo;
+          t.expect(screenOrientationConstantAndroid).toBe(11);
+
+          const desiredOrientations = [
+            ScreenOrientation.Orientation.LANDSCAPE_RIGHT,
+            ScreenOrientation.Orientation.LANDSCAPE_LEFT,
+          ];
+          const validOrientations = [
+            ScreenOrientation.Orientation.LANDSCAPE_RIGHT,
+            ScreenOrientation.OrientationLock.LANDSCAPE_LEFT,
+            ScreenOrientation.Orientation.PORTRAIT_UP,
+          ];
+          await applyAsync({ desiredOrientations, validOrientations });
+        } catch (error) {
+          t.fail(error);
+        }
+      });
+
+      t.it('Apply a native ios lock', async () => {
+        // This test only applies to ios devices
+        if (Platform.OS !== 'ios') {
+          return;
+        }
+
+        try {
+          // Allow only PORTRAIT_UP and LANDSCAPE_RIGHT
+          await ScreenOrientation.lockPlatformAsync({
+            screenOrientationArrayIOS: [
+              ScreenOrientation.Orientation.PORTRAIT_UP,
+              ScreenOrientation.Orientation.LANDSCAPE_RIGHT,
+            ],
+          });
+
+          // detect the correct orientationLock policy immediately
+          const orientationLock = await ScreenOrientation.getOrientationLockAsync();
+          t.expect(orientationLock).toBe(ScreenOrientation.OrientationLock.OTHER);
+
+          // expect the native platform getter to return correctly
+          const platformInfo = await ScreenOrientation.getPlatformOrientationLockAsync();
+          const { screenOrientationArrayIOS } = platformInfo;
+          t.expect(screenOrientationArrayIOS).toContain(
+            ScreenOrientation.Orientation.LANDSCAPE_RIGHT
+          );
+          t.expect(screenOrientationArrayIOS).toContain(ScreenOrientation.Orientation.PORTRAIT_UP);
+
+          const desiredOrientations = [
+            ScreenOrientation.Orientation.PORTRAIT_UP,
+            ScreenOrientation.Orientation.LANDSCAPE_RIGHT,
+          ];
+          const validOrientations = [
+            ScreenOrientation.Orientation.PORTRAIT_UP,
+            ScreenOrientation.Orientation.LANDSCAPE_RIGHT,
+          ];
+          await applyAsync({ desiredOrientations, validOrientations });
+        } catch (error) {
+          t.fail(error);
+        }
+      });
+
+      t.it('Remove all listeners and expect them never to be called', async () => {
+        try {
+          // Register for screen orientation changes
+          let listenerWasCalled = false;
+          ScreenOrientation.addOrientationChangeListener(async () => {
```

Comment:
```suggestion
          ScreenOrientation.addOrientationChangeListener(() => {
```

etc...

In most event listener APIs, the listener system is not set up to accept async functions (errors go uncaught)

Reply:
ive gone ahead and applied this to all the other cases